synapseclient/core/remote_file_storage_wrappers.py

```diff
@@ -117,28 +151,36 @@ def download_file(
 
     @staticmethod
     def upload_file(
-        bucket,
-        endpoint_url,
-        remote_file_key,
-        upload_file_path,
+        bucket: str,
+        endpoint_url: str,
+        remote_file_key: str,
+        upload_file_path: str,
         *,
-        profile_name=None,
-        credentials=None,
-        show_progress=True,
-        transfer_config_kwargs=None,
-    ):
+        profile_name: str = None,
+        credentials: typing.Dict[str, str] = None,
+        show_progress: bool = True,
+        transfer_config_kwargs: dict = None,
+    ) -> str:
         """
         Upload a file to s3 using boto3.
 
-        :param bucket:                  name of bucket to upload to
-        :param endpoint_url:            a boto3 compatible endpoint url
-        :param remote_file_key:         object key to upload the file to
-        :param upload_file_path:        local path of the file to upload
-        :param profile_name:            AWS profile name from local aws config, mutually exclusive with credentials
-        :param credentials:             a dictionary of AWS credentials to use, mutually exclusive with profile_name
-                                        (aws_access_key_id, aws_secret_access_key, aws_session_token)
-        :param show_progress:           whether to print progress indicator to console
-        :param transfer_config_kwargs:  boto S3 transfer configuration (see boto3.s3.transfer.TransferConfig)
+        Arguments:
+            bucket: name of bucket to upload to
+            endpoint_url: a boto3 compatible endpoint url
+            remote_file_key: object key to upload the file to
+            upload_file_path: local path of the file to upload
+            profile_name: AWS profile name from local aws config, mutually exclusive with credentials
```

Comment:
Same comments from `download_file` for some formatting suggestions

Reply:
changed in latest commit